dashboard/public/pixelation/pixelation.js

```diff
@@ -0,0 +1,365 @@
+    document.getElementById("width").value = document.getElementById("widthRange").value = w;
+    document.getElementById("height").value = document.getElementById("heightRange").value = h;
+    binCode = binCode.substring(16, binCode.length);
+
+    if (options.version != '2') {
+      bitsPerPix = binToInt(readByte(binCode, 0));
+      document.getElementById("bitsPerPixel").value = bitsPerPix;
+      document.getElementById("bitsPerPixelSlider").value = bitsPerPix;
+      binCode = binCode.substring(8, binCode.length);
+
+      // Update pixel format indicator.
+      var bitsPerPixel = parseInt(document.getElementById("bitsPerPixel").value);
+      if (hexMode && bitsPerPixel % 4 !== 0) {
+        pixel_format.innerHTML = '<span class="unknown">' + pad('', Math.ceil(bitsPerPixel / 4), '-') + '</span>';
+      } else {
+        if (bitsPerPixel % 3 === 0) {
+          var str;
+          if (hexMode) {
+            str = pad('', bitsPerPixel / 12, 'F');
+          } else {
+            str = pad('', bitsPerPixel / 3, '1');
+          }
+          pixel_format.innerHTML =
+              '<span class="r">' + str + '</span>'
+              + '<span class="g">' + str + '</span>'
+              + '<span class="b">' + str + '</span>';
+        } else {
+          if (hexMode) {
+            pixel_format.innerHTML = pad('', bitsPerPixel / 4, 'F');
+          } else {
+            pixel_format.innerHTML = pad('', bitsPerPixel, '1');
+          }
+        }
+      }
+    }
+  }
+
+  var colorNums = bitsToColors(binCode, bitsPerPix);
+
+  var sqSize = 1, fillSize = 1, offset = 0;
+  if (!document.querySelector('input#actual_size:checked')) {
+    // Auto-size pixel borders and edge offsets.
+    sqSize = MAX_SIZE / Math.max(w, h);
+    fillSize = sqSize * 0.95;
+    offset = (sqSize - fillSize) / 2;
+    if (sqSize - fillSize < 0.33) {
+      fillSize += 1;
+      offset = 0;
+    }
+  }
+
+  // Draw image.
+  var left = Math.floor((MAX_SIZE - w * sqSize) / 2);
+  var top = Math.floor((MAX_SIZE - h * sqSize) / 2);
+  for (var y = 0; y < h; y++) {
+    for (var x = 0; x < w; x++) {
+      ctx.fillStyle = colorNums[(y * w) + x] || "#fdd";
+      ctx.fillRect(left + x * sqSize + offset, top + y * sqSize + offset, fillSize, fillSize);
+    }
+  }
+}
+
+function formatBitDisplay() {
+
+  var theData = pixel_data.value;
+  var chunksPerLine = parseInt(document.getElementById("width").value);
+  var chunkSize = parseInt(document.getElementById("bitsPerPixel").value);
+
+  // If in binary mode.
+  var newBits = formatBits(theData, chunkSize, chunksPerLine);
+  if (newBits != null) {
+    pixel_data.value = newBits;
+  }
+}
+
+function unformatBits() {
+  pixel_data.value = pixel_data.value.replace(/[ \n]/g, "");
+}
+
+/**
+ * Take an unformatted string of bits, place spaces at "chunkSize" offsets (except for the first 3 bytes).
+ */
+function formatBits(bitString, chunkSize, chunksPerLine) {
+
+  var justBits = bitString.replace(/[ \n]/g, "");
```

Comment:
Right, there are two different cases to consider :) so I think it's easiest to just remove spaces and \n, since the previous regex for pixel_data will ensure there are no incorrect non-whitespace characters.